### PR TITLE
Add PostCommit Go github action

### PR DESCRIPTION
**Pull request by Amar3tto**
The following jenkins job has been migrated:

https://ci-beam.apache.org/job/beam_PostCommit_Go
https://ci-beam.apache.org/job/beam_PostCommit_Go_PR

### How has it been tested?
The job has been tested on Self-hosted GKE runners ubuntu 20.04.
The following test cases are passed:

- by adding a comment to the PR

Screenshot from 2023-08-28 14-17-13

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/beam_PostCommit_Go.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.sha || github.head_ref || github.ref }}-${{ github.event.sender.login }}-${{ github.event.schedule }}'
+  cancel-in-progress: true
+
+jobs:
+  beam_PostCommit_Go:
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'workflow_dispatch' ||
+      github.event_name == 'schedule' ||
```
Comment (aleksandr-dudko) [inline .github/workflows/beam_PostCommit_Go.yml:53]: Could it be necessary for us to have the condition "github.event_name == 'schedule'" if the cron is disabled?

Comment (Amar3tto) [reply .github/workflows/beam_PostCommit_Go.yml]: Fixed

File: .github/workflows/beam_PostCommit_Go.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          export_default_credentials: true
+      - name: GCloud Docker credential helper
+        run: |
+          gcloud auth configure-docker us.gcr.io
+      - name: run Go PostCommit script
+        uses: ./.github/actions/gradle-command-self-hosted-action
+        env:
+          USER: github-actions
+        with:
+          gradle-command: :goPostCommit --no-parallel
```
Comment (aleksandr-dudko) [inline .github/workflows/beam_PostCommit_Go.yml:86]: This is just a suggestion, but it's probably better to separate the "--no-parallel" argument from the task ":goPostCommit" and move it to the "arguments" section.

File: .github/workflows/beam_PostCommit_Go.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          arguments: |
+            -Pdocker-repository-root=us.gcr.io/apache-beam-testing/github-actions
```
Comment (aleksandr-dudko) [inline .github/workflows/beam_PostCommit_Go.yml:88]: I'm not sure about the correct answer. But, maybe it is necessary to implement a separate step, like it's done in Jenkins: https://ci-beam.apache.org/view/PostCommit/job/beam_PostCommit_Go/lastStableBuild/consoleFull
 
`./gradlew :sdks:go:container:docker -Pdocker-repository-root=us.gcr.io/apache-beam-testing/jenkins -Pdocker-tag=20230829-143444`

Comment (Amar3tto) [reply .github/workflows/beam_PostCommit_Go.yml:88]: It is not a separate step, it works as an argument.

**Review by aleksandr-dudko — APPROVED** [review]

**Review by andreydevyatkin — APPROVED** [review]
LGTM

**Pull request merged**